packages/core/src/controllers/relayer.ts

```diff
@@ -315,6 +317,7 @@ export class Relayer extends IRelayer {
       if (!this.isConnectionStalled(error.message)) {
         throw e;
       }
+      this.hasExperiencedNetworkDisruption = true;
```

Comment:
Shouldn't it be this because with the way it's written if it's an error other than stalled connection, we'd never flag `hasExperiencedNetworkDisruption` as true
```suggestion
      this.hasExperiencedNetworkDisruption = true;
      if (!this.isConnectionStalled(error.message)) {
        throw e;
      }
```

Reply:
`this.hasExperiencedNetworkDisruption` flag is used to determine how to handle `disconnect` logic. When socket stalls on connect, there is no disconnect flow involved so it isn't really necessary